src/services/server-group/ServerGroupService.ts

```diff
@@ -233,6 +236,26 @@ export class ServerGroupService {
     return this.getStringFromAllMembers(serverGroup, (server) => this.serverService.getServerStatus(server));
   }
 
+  getServerGroupStatus(serverGroup: IServerGroup): ng.IPromise<string> {
+    return this.getServerStatuses(serverGroup).then((statuses) => {
+      let status: string = "DEGRADED";
+      let statusArray: string[] = [];
+      for (let serverKey in statuses) {
+        statusArray.push(statuses[serverKey].toUpperCase());
+      }
```

Comment:
The logic to generate a status array is now duplicated here and `ServerGroupCtrl`. We should expose this as a single method in `ServerGroupService`